app/views/configuration/_ui_2.html.haml

```diff
@@ -81,131 +77,47 @@
-              .form-group
-                %label.col-md-3.control-label
-                  = _('Images')
-                .col-md-8
-                  %ul.list-inline= render_view_buttons(:manageiq_providers_cloudmanager_template, @edit[:new][:views][:manageiq_providers_cloudmanager_template])
-            - if role_allows(:feature => "orchestration_stack_show_list")
-              .form-group
-                %label.col-md-3.control-label
-                  = _('Stacks')
-                .col-md-8
-                  %ul.list-inline= render_view_buttons(:orchestrationstack, @edit[:new][:views][:orchestrationstack])
-            - if role_allows(:feature => "cloud_volume_show_list")
-              .form-group
-                %label.col-md-3.control-label
-                  = _('Volumes')
-                .col-md-8
-                  %ul.list-inline= render_view_buttons(:cloudvolume, @edit[:new][:views][:cloudvolume])
-            - if role_allows(:feature => "auth_key_pair_cloud_show_list")
-              .form-group
-                %label.col-md-3.control-label
-                  = _('Key Pairs')
-                .col-md-8
-                  %ul.list-inline= render_view_buttons(:authkeypaircloud, @edit[:new][:views][:authkeypaircloud])
-            - if role_allows(:feature => "cloud_object_store_container_show_list")
-              .form-group
-                %label.col-md-3.control-label
-                  = _('Object Store')
-                .col-md-8
-                  %ul.list-inline= render_view_buttons(:cloudobjectstorecontainer, @edit[:new][:views][:cloudobjectstorecontainer])
+            - {:manegeiq_providers_cloudmanager => ["ems_cloud_show_list",                    _('Cloud Providers')],
+               :availabilityzone                => ["availability_zone_show_list",            _('Availability Zones')],
+               :cloudtenat                      => ["cloud_tenant_show_list",                 _('Tenants')],
+               :flavor                          => ["flavor_show_list",                       _('Flavors')],
+               :manageiq_providers_cloudmanager_vm       => [["instances_accord",
+                                                              "instances_filter_accord"], _('Instances')],
+               :manageiq_providers_cloudmanager_template => [["images_accord",
+                                                              "images_filter_accord"],    _('Images')],
+               :orchestrationstack        => ["orchestration_stack_show_list",          _('Stacks')],
+               :cloudvolume               => ["cloud_volume_show_list",                 _('Volumes')],
+               :authkeypaircloud          => ["auth_key_pair_cloud_show_list",          _('Key Pairs')],
+               :cloudobjectstorecontainer => ["cloud_object_store_container_show_list", _('Object Store')]}.each do |resource, (feature, view_name)|
+              - if Array(feature).any? {|f| role_allows(:feature => f)}
```

Comment:
so this line is almost what `has_any_role` does.
And the very same line number 115